lib/rules/no-use-computed-property-like-method.js

```diff
@@ -0,0 +1,238 @@
+
+          if (
+            returnStatement.argument.type === 'CallExpression' &&
+            returnStatement.argument.callee.type === 'MemberExpression' &&
+            returnStatement.argument.callee.property.type === 'Identifier'
+          )
+            return {
+              type:
+                propertyMap[returnStatement.argument.callee.property.name]
+                  .valueType.type
+            }
+        }
+
+        /**
+         * Use value as Object even if object includes method
+         */
+        if (
+          property.groupName === 'computed' &&
+          returnStatement.argument.type === 'ObjectExpression'
+        ) {
+          return {
+            type: 'Object'
+          }
+        }
+
+        const evaluated = eslitUtils.getStaticValue(returnStatement.argument)
+
+        if (evaluated) {
+          return {
+            type: getPrototypeType(evaluated.value)
+          }
+        }
+      }
+    }
+
+    const evaluated = eslitUtils.getStaticValue(property.property.value)
+
+    if (evaluated) {
+      return {
+        type: getPrototypeType(evaluated.value)
+      }
+    }
+  }
+  return {
+    type: null
+  }
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow use computed property like method',
+      categories: undefined,
+      url:
+        'https://eslint.vuejs.org/rules/no-use-computed-property-like-method.html'
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      unexpected: 'Use {{ likeProperty }} instead of {{ likeMethod }}.'
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    const GROUP_NAMES = ['data', 'props', 'computed', 'methods']
+    const groups = new Set(GROUP_NAMES)
+
+    /** @type ProprtyMap */
+    const propertyMap = {}
+
+    /**@type ObjectExpression */
+    let vueNodeMap = {}
+
+    return utils.defineVueVisitor(context, {
+      onVueObjectEnter(node) {
+        vueNodeMap = node
+        const properties = utils.iterateProperties(node, groups)
+
+        for (const property of properties) {
+          propertyMap[property.name] = {
+            ...propertyMap[property.name],
+            ...property,
+            valueType: getValueType({ property })
+          }
+        }
+      },
+
+      /**
+       * Re-check propertyMap's valueType.
+       * Because there is a possibility that propertyMap's member reference other propertyMap's member.
+       */
+      'MemberExpression[object.type="ThisExpression"]'(node) {
+        const properties = utils.iterateProperties(vueNodeMap, groups)
+
+        for (const property of properties) {
+          propertyMap[property.name] = {
+            ...propertyMap[property.name],
+            ...property,
+            valueType: getValueType({ property, propertyMap })
+          }
+        }
+      },
+
+      /** @param {ThisExpression} node */
+      'ThisExpression[parent.type="MemberExpression"][parent.parent.type="CallExpression"]'(
```

Comment:
Perhaps you can simplify the selector.

```suggestion
      'CallExpression > MemberExpression > ThisExpression'(
```

Reply:
done
https://github.com/vuejs/eslint-plugin-vue/pull/1234/commits/b06421b1ba134edab580e0379ebf036521a7a7de